From 76e466235bcf50d81d59f122aee19b3652c5e6db Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Tue, 28 Aug 2018 20:31:16 +0300
Subject: [PATCH 15/28] soong_config: Add flag for crypto waiting on QSEE to
 start

 * Every single new device supports this logic except
   a few ancient ones and nexus.

Change-Id: I9a5493667c6bb3636468a90df0da65cf5e8c3a8b
---
 build/soong/android/variable.go | 4 ++++
 build/soong/soong_config.mk     | 1 +
 2 files changed, 5 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index a3ce7f6e..79bf9473 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -6,6 +6,9 @@ type Product_variables struct {
 	Has_legacy_camera_hal1 struct {
 		Cflags []string
 	}
+	Should_skip_waiting_for_qsee struct {
+		Cflags []string
+	}
 	Supports_hw_fde struct {
 		Cflags []string
 		Header_libs []string
@@ -42,6 +45,7 @@ type ProductVariables struct {
 	Additional_gralloc_10_usage_bits  *string `json:",omitempty"`
 	Has_legacy_camera_hal1  *bool `json:",omitempty"`
 	Java_Source_Overlays *string `json:",omitempty"`
+	Should_skip_waiting_for_qsee  *bool `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
 	Supports_hw_fde  *bool `json:",omitempty"`
 	Supports_hw_fde_perf  *bool `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 4083edf7..342b322e 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -6,6 +6,7 @@ _contents := $(_contents)    "Lineage":{$(newline)
 $(call add_json_str_omitempty, Additional_gralloc_10_usage_bits, $(TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS))
 $(call add_json_bool, Has_legacy_camera_hal1, $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
 $(call add_json_str, Java_Source_Overlays, $(JAVA_SOURCE_OVERLAYS))
+$(call add_json_bool, Should_skip_waiting_for_qsee, $(filter true,$(TARGET_KEYMASTER_SKIP_WAITING_FOR_QSEE)))
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
 $(call add_json_bool, Supports_hw_fde, $(filter true,$(TARGET_HW_DISK_ENCRYPTION)))
 $(call add_json_bool, Supports_hw_fde_perf, $(filter true,$(TARGET_HW_DISK_ENCRYPTION_PERF)))
-- 
2.17.1

